### PR TITLE
Split code into a submodules

### PR DESCRIPTION
**Pull request by p-snft**
This is refactoring only. Every change in this PR should be fully transparent and invisible to the users. In the end, every class should sit in their own file while the API is unchanged.

Implements #751.

### CODE REVIEW

**Comment by p-snft** [comment]
The problem detected by the pipelines is due to Black and isort not agreeing on formating imports.

The decreased coverage seems to be an artifact, as nothing really changed.

**Comment by p-snft** [comment]
* The decreased coverage seems to be an artifact, as nothing really changed.
* Codacity dislikes unused imports in the __init__ files.

The second problem is actually not a problem, so I would like to merge.

**Comment by p-snft** [comment]
Todo:

- [ ] Update Docs
- [ ] Adjust copyright notes (?)

File: src/oemof/solph/blocks/bus.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+"""Creating sets, variables, constraints and parts of the objective function
+for Bus objects.
+
+SPDX-FileCopyrightText: Uwe Krien <krien@uni-bremen.de>
+SPDX-FileCopyrightText: Simon Hilpert
+SPDX-FileCopyrightText: Cord Kaldemeyer
+SPDX-FileCopyrightText: Patrik Schönfeldt
+SPDX-FileCopyrightText: Birgit Schachler
+SPDX-FileCopyrightText: jnnr
+SPDX-FileCopyrightText: jmloenneberga
```
Comment (p-snft) [inline src/oemof/solph/blocks/bus.py:12]: These copyright notes are taken from blocks.py. Some of these authors might not have contributed to the new blocks/bus.py. Should we adjust this? (Might be tedious in some cases.)

**Review by uvchik — APPROVED** [review]
You changed the structure as decided at the developer meeting. It is a good idea to merge the API-stable changes in a first PR. Thnak you. Go on :smile:

**Comment by simnh** [comment]
I think its good!

**Comment by jokochems** [comment]
> 
> 
> I think its good!

Yeah, me too. Thanks for making the structure of solph clearer and more readable.

**Comment by uvchik** [comment]
I just found out that the references to the docstrings in the documentation are all broken now.

**Comment by p-snft** [comment]
Can you give an example? There is a test and I fixed a lot of references.

**Pull request merged**